regression-tests/src/helpers/selenium.helpers.ts

```diff
@@ -42,20 +96,155 @@ export const initializeSeleniumDriver = async (browserType) => {
+		results.push({ textElement, closerElement });
+	});
+
+	return results;
+};
+
+export const findElementNearText = async (driver: WebDriver, text:string, tag: string) => {
+	const result = (await findElementsNearText(driver, text, tag))[0];
+	return result?.closerElement;
+};
+
+export const findInputNearText = async (driver: WebDriver, text:string) => findElementNearText(driver, text, 'input');
+
+export const fillInputByLabel = async (driver: WebDriver, label, value) => {
+	const input = await findInputNearText(driver, label);
+	await driver.wait(until.elementIsVisible(input));
+	await input.sendKeys(value);
+};
+
+export const waitForPageToBeLoaded = async (driver: WebDriver) => {
+	await driver.wait(until.elementLocated(By.css('body')), 100000);
+	await animationsEnded(driver);
 };
 
 export const navigateTo = async (driver:WebDriver, page:string) => {
 	await driver.get(getUrl(page));
-	await driver.wait(until.elementLocated(By.css('body')), 100000);
+	await waitForPageToBeLoaded(driver);
+};
+
+export const closeOriginWindow = async (driver: WebDriver) => {
+	const windows = await (driver as WebDriver).getAllWindowHandles();
+	await driver.switchTo().window(windows[0]);
+	await driver.close();
+	await driver.switchTo().window(windows[1]);
+};
+
+export const clickOn = async (driver: WebDriver, buttonContent:string, elementNumber?:number) => {
+	await waitUntilPageLoaded(driver);
+	const text = `//body//*[text()="${buttonContent}"]|//body//input[@value='${buttonContent}']`;
+	const target = By.xpath(text);
+	const elements = await getElements(driver, target);
+
+	let link;
+
+	if (elementNumber) {
+		link = elements[elementNumber - 1]; // element number starts from 1
+	} else {
+
+		const tagOrder = ['a', 'button', 'input', 'div'];
+		const getTagPriority = (tag):number => tagOrder.includes(tag) ? tagOrder.indexOf(tag) : tagOrder.length;
+
+		await Promise.all(elements.map(async (element) => {
+			(element as any).tag = await element.getTagName();
+		}));
+		
+		link = elements.sort((a:any, b:any) => getTagPriority(a.tag) - getTagPriority(b.tag)) [0];
+	}
+
+	await driver.wait(until.elementIsEnabled(link));
+	await link.click();
+	await animationsEnded(driver);
+};
+
+export const fillInForm = async (driver: WebDriver, fields: Record<Label, Value>) => {
+	await waitUntilPageLoaded(driver);
+	try {
+		await Promise.all(Object.keys(fields).map((labelName)=> fillInputByLabel(driver, labelName, fields[labelName])));
+	} catch (e) {
+		// there is a tolarance on the label in order to reuse the same table
+		console.warn('Warning when trying to fillInForm');
+		console.warn(e);
+	}
+
 };
+
+export const clickOnCheckboxNearText = async (driver: WebDriver, text: string) => {
+	await animationsEnded(driver);
+	const checkbox = await findInputNearText(driver, text);
+	await driver.wait(until.elementIsEnabled(checkbox), 100000);
+	await checkbox.click();
+};
+
+
+const buttonsPaths = {
+	'avatar': '(//button)[4]',
```

Comment:
yeah beccause it could match another one. this is more like the avatar menu that I now is there. If there are problems in the future regarding this we can review it.